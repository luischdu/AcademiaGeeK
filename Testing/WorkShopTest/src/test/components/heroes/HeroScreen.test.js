import { mount } from 'enzyme'
import '@testing-library/jest-dom'
import React from 'react'
import { MemoryRouter } from 'react-router'
import { HeroScreen } from '../../../components/heroes/HeroScreen'
import { Route } from 'react-router-dom'

describe('Validacines en el componente <HeroScreen />', () => {

    const historyMock = {
        push: jest.fn(),
        goBack: jest.fn(),
        length: 10
    }

    const wrapper = mount(
        <MemoryRouter initialEntries={['/hero']}>
            <HeroScreen history={historyMock} />
        </MemoryRouter>
    )

    test('Redirect si no hay URL', () => {
        expect(wrapper.find('Redirect').exists()).toBe(true)
    })
    
    
    test('Funcionamiento correcto', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-hulk']}>
                <Route path='/hero/:heroeId' component={HeroScreen} />
            </MemoryRouter>
        )

        expect(wrapper).toMatchSnapshot()
        
        const textContent = wrapper.find('h3').text().trim()

        expect(textContent).toBe('Hulk')
    })
    
})