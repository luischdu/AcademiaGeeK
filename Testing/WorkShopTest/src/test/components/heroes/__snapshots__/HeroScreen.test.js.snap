// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validacines en el componente <HeroScreen /> Funcionamiento correcto 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Hulk"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="../assets/heroes/marvel-hulk.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
       
      Hulk
       
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
         
        <b>
           Alter ego: 
        </b>
         
        Bruce Banner
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
           Publisher: 
        </b>
         
        Marvel Comics
         
      </li>
      <li
        className="list-group-item"
      >
         
        <b>
           First appearance: 
        </b>
         
        The Incredible Hulk #1
         
      </li>
    </ul>
    <h5>
       Characters 
    </h5>
    <p>
       
      Bruce Banner
       
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
</div>
`;
